### ISSUE TITLE
There is invalid type in PHPDoc block of \Magento\Framework\Data\Tree::getNodeById()

### ISSUE THREAD

**Issue by andkirby** [magento/magento2#8507]
Here is a minor issue which breaks code validation.
It looks like param `$nodeId` has wrong type `Node` in PHPDoc of method `\Magento\Framework\Data\Tree::getNodeById()`.
```
    /**
     * Enter description here...
     *
     * @param Node $nodeId
     * @return Node
     */
    public function getNodeById($nodeId)
    {
        return $this->_nodes->searchById($nodeId);
    }
```
Seem, it can be `string|int`.

And `Enter description here...` says nothing about. :)

**Comment by magento-engcom-team** [comment]
@andkirby, thank you for your report.
We've created internal ticket(s) MAGETWO-84247 to track progress on the issue.

**Comment by okorshenko** [comment]
Hi @andkirby 
The issue has been fixed and delivered in 2.2-develop branch. Will be available with upcoming patch release.
